### PR TITLE
refactor: Remove `DataSet` aliases and mentions

### PR DESCRIPTION
**Pull request by merelcht**
## Description
Closes [Remove all old DataSet classes#3159](https://github.com/kedro-org/kedro/issues/3159)

## Development notes
- Removed all aliases from `DataSet` to `Dataset`
- Renamed all remaining mentions of `DataSet` to `Dataset`
- Removed mentions of `polars.GenericDataset` which is replaced by `polars.EagerPolarsDataset`

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: kedro-datasets/docs/source/kedro_datasets.rst [hunk 1]
```diff
@@ -11,97 +11,52 @@ kedro_datasets
    :toctree:
    :template: autosummary/class.rst
 
-   kedro_datasets.api.APIDataSet
```
Comment (astrojuanlu) [inline kedro-datasets/docs/source/kedro_datasets.rst:14]: So happy to see these go away!

File: kedro-datasets/kedro_datasets/api/__init__.py [hunk 2]
```diff
@@ -9,9 +9,8 @@
 import lazy_loader as lazy
 
 # https://github.com/pylint-dev/pylint/issues/4300#issuecomment-1043601901
-APIDataSet: type[APIDataset]
 APIDataset: Any
```
Comment (astrojuanlu) [inline kedro-datasets/kedro_datasets/api/__init__.py:12]: Is there anything we can simplify or optimise in these `__init__.py` files now that the aliases are gone @deepyaman ?

**Review by SajidAlamQB — APPROVED** [review]
Amazing!! 🌟 Feels like a weight lifted off our shoulders!! 🚀🚀🎉🎉

File: kedro-datasets/setup.py [hunk 3]
```diff
@@ -15,54 +15,50 @@ def _collect_requirements(requires):
...
+pillow_require = {"pillow.ImageDataset": ["Pillow~=9.0"]}
 plotly_require = {
-    "plotly.PlotlyDataSet": [PANDAS, "plotly>=4.8.0, <6.0"],
-    "plotly.JSONDataSet": ["plotly>=4.8.0, <6.0"],
+    "plotly.PlotlyDataset": [PANDAS, "plotly>=4.8.0, <6.0"],
+    "plotly.JSONDataset": ["plotly>=4.8.0, <6.0"],
 }
 polars_require = {
-    "polars.CSVDataSet": [POLARS],
-    "polars.GenericDataSet":
```
Comment (AhdraMeraliQB) [inline kedro-datasets/setup.py:62]: What happened to `polars.GenericDataSet`?

Comment (merelcht) [reply kedro-datasets/setup.py:62]: It's replaced by `polars.EagerPolarsDataset`.

Comment (AhdraMeraliQB) [reply kedro-datasets/setup.py:62]: Right, I see it's in the PR description too 🤦 😂

**Review by AhdraMeraliQB — APPROVED** [review]
Great to see these go 🥳

File: kedro-datasets/kedro_datasets/api/__init__.py (hunk 2 above)
Comment (merelcht) [reply kedro-datasets/kedro_datasets/api/__init__.py:10]: @astrojuanlu the `__init__.py` files are now back to how they were before all the aliasing.

**Pull request merged**